### PR TITLE
Better management of api key

### PR DESCRIPTION
**Pull request by kalkov**
Added a new API key sub-menu.
It contains one item "Add" if there is no API key. Or "View", "Validate", "Remove" if there is a saved key. 

"View" displays an obfuscated key: 
Screenshot from 2019-05-16 16-15-27

"Validate" displays a message: 
Screenshot from 2019-05-16 16-19-00

"Remove" asks for confirmation:
Screenshot from 2019-05-16 16-19-52
And displays a confirmation message after successful action: 
Screenshot from 2019-05-16 16-20-53

### CODE REVIEW

**Comment by IvanIvanoff** [comment]
The API key works by taking the first part (before the `_` - it's some randomly generated string) and calculates HMAC with a secret known only by the server (stored encrypted in the devops repo). So it's safe to display the whole first part - it is even written to the logs and stored plain text in DB because it's not enough to generate the whole api key.

Also it may be beneficial to show the first part as we can easily search the logs for it.

On the other hand, the user can see the whole API key in the sanbase's account page. Why hide it here?

**Comment by valo** [comment]
yeah, but if you do screensharing/recording and the key is revealed in plaintext, somebody can use that API key to fetch data on your behalf. I think it is better to not show the API key completely.

File: src/apiKey.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 var API_KEY = 'API_KEY'
 var API_KEY_LOG_TYPE = 'ApiKeyLog'
+var ADD_API_KEY_ACTION = 'AddApiKey'
+var VALIDATE_API_KEY_ACTION = 'ValidateApiKey'
```
Comment (IvanIvanoff) [inline src/apiKey.js]: should there be `REMOVE_API_KEY_ACTION` also?

Comment (kalkov) [reply src/apiKey.js]: Good point, added logging when a user removes an API key.

File: src/.eslintrc.json [hunk 2]
```diff
@@ -28,16 +28,21 @@
...
     "assertCanAccessHistoricData_": "readonly",
     "assertHasData_": "readonly",
-    "setUserProperty_": "readonly",
     "apiKeyProperty_": "readonly",
     "addApiKey_": "readonly",
+    "removeApiKey_": "readonly",
+    "deleteApiKeyProperty_": "readonly",
     "hasApiKeyProperty_": "readonly",
+    "obfuscateApiKey_": "readonly",
+    "checkApiKeyStillValid_": "readonly",
```
Comment (IvanIvanoff) [inline src/.eslintrc.json]: why`still` valid? Is there another function that checks the validity and then this function/property? starts to handle/persist the checks?

Comment (kalkov) [reply src/.eslintrc.json]: One checks

**Pull request merged**